rasa/core/policies/ensemble.py

```diff
@@ -91,590 +220,318 @@ def _check_for_important_policies(self) -> None:
-                except TypeError as e:
-                    raise Exception(f"Could not initialize {policy_name}. {e}") from e
-                parsed_policies.append(policy_object)
-            except (ImportError, AttributeError) as e:
-                raise InvalidPolicyConfig(
-                    f"Module for policy '{policy_name}' could not "
-                    f"be loaded. Please make sure the "
-                    f"name is a valid policy."
-                ) from e
-
-        return parsed_policies
-
-    @classmethod
-    def get_featurizer_from_dict(cls, policy: Dict[Text, Any]) -> Tuple[Any, Any]:
-        """Gets the featurizer initializer and its arguments from a policy config."""
-        # policy can have only 1 featurizer
-        if len(policy["featurizer"]) > 1:
-            raise InvalidPolicyConfig(
-                f"Every policy can only have 1 featurizer "
-                f"but '{policy.get('name')}' "
-                f"uses {len(policy['featurizer'])} featurizers."
-            )
-        featurizer_config = policy["featurizer"][0]
-        featurizer_name = featurizer_config.pop("name")
-        featurizer_func = registry.featurizer_from_module_path(featurizer_name)
-
-        return featurizer_func, featurizer_config
+    def validate(self, ensemble: List[Policy], domain: Domain) -> None:
+        """Checks whether this ensemble can be applied to the given policies.
 
-    @classmethod
-    def get_state_featurizer_from_dict(
-        cls, featurizer_config: Dict[Text, Any]
-    ) -> Tuple[Callable, Dict[Text, Any]]:
-        # featurizer can have only 1 state featurizer
-        if len(featurizer_config["state_featurizer"]) > 1:
-            raise InvalidPolicyConfig(
-                f"Every featurizer can only have 1 state "
-                f"featurizer but one of the featurizers uses "
-                f"{len(featurizer_config['state_featurizer'])}."
-            )
-        state_featurizer_config = featurizer_config["state_featurizer"][0]
-        state_featurizer_name = state_featurizer_config.pop("name")
-        state_featurizer_func = registry.state_featurizer_from_module_path(
-            state_featurizer_name
-        )
-
-        return state_featurizer_func, state_featurizer_config
-
-
-class SimplePolicyEnsemble(PolicyEnsemble):
-    """Default implementation of a `Policy` ensemble."""
+        Args:
+          ensemble: a list of policies
+          domain: the common domain
+        Raises:
+          `InvalidPolicyEnsembleConfig`: if this ensemble cannot be applied to
+            the given ensemble of policies
+        """
+        super().validate(ensemble=ensemble, domain=domain)
+        # TODO: shouldn't we raise in case priorities are not unique since this might
+        # lead to unexpected results in edge cases (swap order in your config)
+        self.warn_priorities_not_unique(ensemble=ensemble)
+        # TODO: shouldn't we assert that there's at most one rule policy here?
```

Comment:
Not sure. A customer might build their own special rule policy that can work in parallel with ours.

Reply:
I added this because `rule_only_data` is picked from the first rule policy that is found which seemed odd. -- But I can remove this and we'll add a note in the rulepolicy or whatever is distributing that rule_policy. (FYI @wochinge )

Reply:
Let's keep it out of this PR and focus on migrating the ensemble in this PR. I'd not even create a note - if even then an issue.